src/process-events.c

```diff
@@ -148,15 +148,6 @@ struct bpf_map_def SEC("maps/tail_call_table") tail_call_table = {
         read_value(ts, CRC_TASK_STRUCT_REAL_PARENT, &ptr, sizeof(ptr));                                 \
         read_value(ptr, CRC_TASK_STRUCT_TGID, &ppid, sizeof(ppid));                                     \
         read_value(ts, CRC_TASK_STRUCT_LOGINUID, &luid, sizeof(luid));                                  \
-        read_value(ts, CRC_TASK_STRUCT_MM, &ptr, sizeof(ptr));                                          \
-        read_value(ptr, CRC_MM_STRUCT_EXE_FILE, &ptr, sizeof(ptr));                                     \
-        SET_OFFSET(CRC_FILE_F_PATH);                                                                    \
-        ptr = ptr + *(u32 *)offset; /* ptr to f_path */                                                 \
-        read_value(ptr, CRC_PATH_DENTRY, &ptr, sizeof(ptr));                                            \
-        SET_OFFSET(CRC_DENTRY_D_NAME);                                                                  \
-        ptr = ptr + *(u32 *)offset; /* ptr to d_name */                                                 \
-        read_value(ptr, CRC_QSTR_LEN, &length, sizeof(length));                                         \
-        read_value(ptr, CRC_QSTR_NAME, &exe, sizeof(exe));                                              \
```

Comment:
Hmm, it looks like this ought to retrieve the name associated with the file that's retrieved via `/proc/[pid]/exe`.  I wonder why we weren't using this?

Reply:
I think there's a couple of gotchas but maybe we should:

1. We want the entire path to the executable. This used to get just the name.
2. We were doing this for *every* syscall which is a waste, only execve* really needs to know the exec path
3. At the time call for this we are still in the executable prior to the syscall finishing. In the case of execve* (the ones were we actually need this information) the exe name here would be for the executable we are replacing not the one we are launching. 

All those gotchas can be addressed though I think:

1. We can use the function I pulled out in the refactor to trace the `path` back step by step to get the full path.
2. We can have it just be used for the `execveat` syscall. The reason to not do it for `execve` is so we can still tell if a executable was run using a relative or absolute path
3. We can do this in the `kretprobe` so the `execveat` has finalized and the image has been replaced. 

I think this may actually be better than adding that new program that hooks into that random function. Now I feel silly not thinking of doing that when I was deleting this code. I can give it a shot.

Reply:
So, I didn't see in the diffs where we got the other file identity pieces (inode, dev major and minor), but they're also available via the `task->mm->exe_file->f_path->dentry` route. Apparently the `f_path` value also has a pointer to the `vfsmount` structure, which might have some fields that could help us find the path in the filesystem, though I'm not entirely sure about how that would work.

Anyway, I didn't see anything that jumped out at me as concerning, and I really liked the examples you set up in the PR description.